File: python/228_汇总区间_easy.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if not nums:
            return nums
        i = 0
        res = []
        while i<len(nums):
            start = end = nums[i]
            while i+1<len(nums) and nums[i+1] == end+1:
                end += 1
                i += 1
            res.append(str(start) if start==end else str(start)+'->'+str(end))
            i += 1
        return res

File: cpp/228_汇总区间_easy.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        int N = nums.size();
        if(!N)
            return {};
        int i=0;
        vector<string> res;
        while(i<N){
            int start = nums[i];
            int end = start;
            while(i+1<N && nums[i+1]==end+1){
                i++;
                end++;
            }
            if(start == end)
                res.push_back(to_string(start));
            else{
                string tmp = to_string(start);
                tmp.append("->");
                tmp.append(to_string(end));
                res.push_back(tmp);
            }
            i++;
        }
        return res;
    }
};